algorithms/kernel/kernel_function/kernel_function_linear_batch_container.h

```diff
@@ -65,8 +76,19 @@ services::Status BatchContainer<algorithmFPType, method, cpu>::compute()
 
     ComputationMode computationMode = static_cast<ParameterBase *>(par)->computationMode;
 
-    __DAAL_CALL_KERNEL(env, internal::KernelImplLinear, __DAAL_KERNEL_ARGUMENTS(method, algorithmFPType), compute, computationMode, a[0], a[1], r[0],
-                       par);
+    auto & context    = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & deviceInfo = context.getInfoDevice();
+
+    if (method == defaultDense && !deviceInfo.isCpu)
+    {
+        __DAAL_CALL_KERNEL_SYCL(env, internal::KernelImplLinearOneAPI, __DAAL_KERNEL_ARGUMENTS(method, algorithmFPType), compute, computationMode,
+                                *a[0], *a[1], *r[0], par);
+    }
+    else
+    {
+        __DAAL_CALL_KERNEL(env, internal::KernelImplLinear, __DAAL_KERNEL_ARGUMENTS(method, algorithmFPType), compute, computationMode, a[0], a[1],
+                           r[0], par);
```

Comment:
Inconsistent: for CPU kernel, `a` and `r` are passed as references, but for GPU kernel as pointers